jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -369,9 +376,10 @@ private final long memSegmtToLongArg(MemorySegment argValue) throws IllegalState
 			 * is captured so as to reset the internal index and avoid retaining the references
 			 * to the unreachable objects.
 			 */
-			HeapArgInfo info = heapArgInfo.get();
-			if (info != null) {
-				info.clear();
+			Deque<HeapArgInfo> infoStack = heapArgInfo.get();
+			if(!flagCreateHeapArgInfo.get() && !infoStack.isEmpty()) {
```

Comment:
Same applies here, the second check can be removed and please add a space after if.